### PR TITLE
Fix System Pin Feature Gate

### PR DESCRIPTION
**Pull request by martinmortsell**
This pull request addresses #37 .

Instead of allowing usage of system pins (which fails) without the `reconfigurable-system-pins` feature set, not having the feature set now removes the pins completely, and any peripheral that directly depend on them.

It would be nice to achieve compilation failures that are more descriptive than
```
error[E0609]: no field `pb4` on type `BankB`
  --> examples/blinky.rs:62:25
   | 
62 |         let pb4 = bankb.pb4.into_output(true);
   |                         ^^^ help: a field with a similar name exists: `pb0`
```
but I have no idea of how to achieve that.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 ### Fixed
 
 - HAL build when targeting `sams70n19b`.
+- reconfigurable-system-pins feature.
```
Comment (tmplt) [inline CHANGELOG.md]: I suggest adding some details.

File: hal/src/serial/twi.rs [hunk 2]
```diff
@@ -41,11 +41,13 @@ twi.write_read(0x0, &[0b1000_0000], &mut buffer).unwrap();
 use crate::clocks::{Clock, Hertz, HostClock, PeripheralIdentifier};
 use crate::ehal::blocking;
 use crate::generics;
+#[cfg(feature = "reconfigurable-system-pins")]
+use crate::pac::TWIHS1;
```
Comment (tmplt) [inline hal/src/serial/twi.rs:49]: Why is this gated?

File: hal/src/serial/usart/mod.rs [hunk 3]
```diff
@@ -83,15 +83,17 @@ let _ = spi.read().unwrap();
 pub use super::uart::{ChannelMode, ParityMode, UartConfiguration};
 use crate::clocks::{Clock, Hertz, HostClock, Pck, Pck4, PeripheralIdentifier};
 use crate::generics::{self, Token};
+#[cfg(feature = "reconfigurable-system-pins")]
+use crate::pac::USART1;
```
Comment (tmplt) [inline hal/src/serial/usart/mod.rs:91]: Ditto.

**Review by tmplt — CHANGES_REQUESTED** [review]

File: hal/src/serial/twi.rs (hunk 2 above)
Comment (tmplt) [reply hal/src/serial/twi.rs:49]: Nevermind. It's because the peripheral requires the system pins.

**Comment by tmplt** [comment]
Looks logical from a cursory review. I'd double-check if we somehow break API here and lose any read-only features. Add some details to the `CHANGELOG.md` and see the commit history on how to format these; I'd like to keep them uniform. These changes can be squashed into a single commit.

Adding documentation that these peripherals are unavailable without the feature would also be good.

**Comment by martinmortsell** [comment]
Should the CI be updated to run both with and without the `reconfigurable-system-pins` feature?

**Comment by tmplt** [comment]
Yes; a feature matrix may be applicable.

**Comment by michalfita** [comment]
@martinmortsell Please fix issues (merge back `development` shall help) and we can merge this and release 0.4.1.

**Comment by martinmortsell** [comment]
My attempt to add a featrues matrix to the CI seems to have been unsuccesful, reattempting later.

**Comment by martinmortsell** [comment]
CIFeatures
And there you have it.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -12,7 +12,7 @@ jobs:
       - id: boards
         run: echo "::set-output name=board_matrix::$(ls ./boards --indicator-style=none | jq -ncR '[inputs]')"
       - id: features
-        run: echo "::set-output name=feature_matrix::["reconfigurable-system-pins","usart-spi-host-without-select"]"
+        run: echo "::set-output name=feature_matrix::[\"reconfigurable-system-pins\",\"usart-spi-host-without-select\"]"
```
Comment (martinmortsell) [inline .github/workflows/build.yml:15]: Having the features hard coded like this isn't pretty, but I have no idea how to parse the actual list of features and not catch the internal-only features.

File: CHANGELOG.md [hunk 5]
```diff
@@ -24,6 +24,8 @@
 ### Fixed
 
 - HAL build when targeting `sams70n19b`.
+- The special-on-reset pins `PB4/5/6/7/12` are now completely unavailable, if the `reconfigurable-system-pins` feature is not enabled.
+- `Usart<Usart1>` and `Twi<TwiHS1>` are now only available if the `reconfigurable-system-pins` feature is enabled, as they cannot obtain a valid pin configuration without it.
```
Comment (tmplt) [inline CHANGELOG.md]: Move these under `Changed`.

File: .github/workflows/build.yml (hunk 4 above)
Comment (tmplt) [reply .github/workflows/build.yml:15]: There is probably a better way to check different feature permutations (and maybe streamline the CI further), but we can delay that until we have more feature gates. Ideally they will be de-coupled, so we could just check with `--all-features`, but I don't know if we can separate the PACs from this.

**Review by tmplt — CHANGES_REQUESTED** [review]
Fixup the changelog and clean up the commit history and I'll approve. Thanks!

File: .github/workflows/build.yml (hunk 4 above)
Comment (michalfita) [reply .github/workflows/build.yml:15]: There a cargo extensions crate called [cargo-hack](https://github.com/taiki-e/cargo-hack) that can do feature permutations.

Raised #42 about this.

**Comment by michalfita** [comment]
@martinmortsell I can't push to this branch myself. You have to not only move the section to `## Changed` but up to `# Unreleased`, as `v0.4.0` has sailed already.

**Comment by tmplt** [comment]
Changes LGTM. Squash into a single commit to keep the history tidy.

**Comment by martinmortsell** [comment]
> Changes LGTM. Squash into a single commit to keep the history tidy.

But is the squash not done as part of the merge?

**Comment by tmplt** [comment]
> But is the squash not done as part of the merge?

Yes, but I do not have merging rights, so I can only state my wishes.

**Comment by martinmortsell** [comment]
Ok, fair.

**Pull request merged**